src/neptune_scale/sync/operations_repository.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+
+__all__ = ("OperationsRepository", "OperationType", "Operation", "Metadata")
+
+import contextlib
+import os
+import sqlite3
+import threading
+import time
+import typing
+from contextlib import AbstractContextManager
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import (
+    Optional,
+    Union,
+)
+
+from neptune_api.proto.neptune_pb.ingest.v1.common_pb2 import Run as CreateRun
+from neptune_api.proto.neptune_pb.ingest.v1.common_pb2 import UpdateRunSnapshot
+
+from neptune_scale.util import get_logger
+
+logger = get_logger()
+
+DB_VERSION = "v1"
+
+SequenceId = typing.NewType("SequenceId", int)
+
+
+class OperationType(IntEnum):
+    UPDATE_SNAPSHOT = 0
+    CREATE_RUN = 1
+
+
+@dataclass(frozen=True)
+class Operation:
+    sequence_id: SequenceId
+    timestamp: int
+    operation_type: OperationType
+    operation: Union[UpdateRunSnapshot, CreateRun]
+    metadata_size: int
+
+
+@dataclass(frozen=True)
+class Metadata:
+    version: str
+    project: str
+    run_id: str
+    parent: Optional[str] = None
```

Comment:
This needs to be renamed - as it is I don't know if it's a run_id of the parent or an experiment name.

Reply:
https://github.com/neptune-ai/neptune-client-scale/blob/dd12d23bb9ef65449143cea6d28f014828fbcb58/src/neptune_scale/sync/operations_repository.py#L52